File: java/statePattern/FireRockets.java
package statePattern;

public class FireRockets implements State {
	private MissionControl missionControl;
	
	public FireRockets(MissionControl missionControl) {
		this.missionControl = missionControl;
	}

	@Override
	public void checkAllSystems() {
		System.out.println("Systems have been cleared");
	}

	@Override
	public void startPowerUnits() {
		System.out.println("Power units have been started");

	}

	@Override
	public void retractArms() {
		System.out.println("Arms and fuel connections have been retracted.");
		this.missionControl.setState(this.missionControl.getRocketBooster());
	}

	@Override
	public void shutDown() {
		System.out.println("All systems are to cease operations immediatly");
		this.missionControl.setState(this.missionControl.getAbortProcedure());
	}

	@Override
	public void rocketBooster() {
		System.out.println("Power units have not been started.");

	}

}

File: php/statePattern/FireRockets.php
<?php namespace php\statePattern;

class FireRockets implements State {
	
	/**
	 * Mission control instance
	 *
	 * @var MissionControl $missionControl
	 */
	public $missionControl;
	
	public function __construct(MissionControl $missionControl) {
		$this->missionControl = $missionControl;
	}
	
	/**
	 * Check all systems are stable and ready to go
	 *
	 * @return void
	 */
	public function checkAllSystems() {
		echo "Systems have already been cleared.<br>";
	}
	
	/**
	 * print statement for power units
	 *
	 * @return void
	 */
	public function startPowerUnits() {
		echo "Systems have already been cleared.<br>";
	}
	
	/**
 	 * Print  power unit statement
	 *
	 * @return void
	 */
	public function retractArms() {
		echo "Arms and fuel connections have been retracted.<br>";
		$this->missionControl->setState($this->missionControl->getRocketBooster());
	}
	
	/**
 	 * Print shutdown statement
	 *
 	 * @return void
	 */
	public function shutDown() {
		echo "All systems are to cease operations immediatly.<br>";
		$this->missionControl->liftOffHandOver($this->missionControl->getAbortProcedure());
	}
	
	/**
	 * Print negative rocket booster statement
	 *
	 * @return void
	 */
	public function rocketBooster() {
		echo "power units have not been started.<br>";
	}
}

